pkg/buildah/buildah.go

```diff
@@ -0,0 +1,620 @@
+package buildah
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"os/user"
+	"path"
+	"strings"
+
+	dockertypes "github.com/docker/docker/api/types"
+
+	"github.com/openshift/source-to-image/pkg/api"
+	"github.com/openshift/source-to-image/pkg/api/constants"
+	"github.com/openshift/source-to-image/pkg/docker"
+	s2ierr "github.com/openshift/source-to-image/pkg/errors"
+	s2itar "github.com/openshift/source-to-image/pkg/tar"
+	"github.com/openshift/source-to-image/pkg/util/fs"
+	"github.com/openshift/source-to-image/pkg/util/interrupt"
+	utillog "github.com/openshift/source-to-image/pkg/util/log"
+)
+
+var log = utillog.StderrLog
+
+// Buildah implements docker.Docker interface using buildah as a backend.
```

Comment:
OK this is good / was going to look for that.

That said, as a result, I would have expected a buildah_test.go here with some unit tests a la https://github.com/openshift/source-to-image/blob/master/pkg/docker/docker_test.go

Reply:
Since the unit-tests of this component would require shell-out to call `buildah`, I end moving those tests [here](https://github.com/otaviof/source-to-image/blob/oci/test/integration/buildah/buildah_test.go). Therefore, we can make sure in that level of testing we can enable `buildah`, and find it installed.

Reply:
In addition to the actual callouts you have in the integration test you noted, your unit test version of buildah_test.go could have a `fakeBuildah` just like the `fakeDocker` we have at https://github.com/openshift/source-to-image/blob/master/pkg/docker/docker_test.go#L145-L147, no?